src/types/rtti.rs

```diff
@@ -1417,6 +1417,36 @@ impl ISerializable {
             ) as *const Class)
         }
     }
+
+    #[inline]
+    pub fn is_a<U>(&self) -> bool
+    where
+        U: ScriptClass,
+    {
+        self.class()
+            .base_iter_with_self()
+            .any(|class| class.name() == CName::new(U::NAME))
+    }
+
+    #[inline]
+    pub fn is_exactly_a<U>(&self) -> bool
+    where
+        U: ScriptClass,
+    {
+        self.class().name() == CName::new(U::NAME)
+    }
+
+    #[inline]
+    pub fn inner_ref<U>(&self) -> WeakRef<U>
+    where
+        U: ScriptClass,
+    {
+        if !self.is_a::<U>() {
+            return WeakRef::default();
+        }
+        unsafe { mem::transmute::<&red::WeakHandle<red::ISerializable>, &WeakRef<U>>(&self.0.ref_) }
+            .clone()
+    }
```

Comment:
I think this should return an `Option` to indicate that a check has failed
```suggestion
    pub fn inner_ref<U>(&self) -> Option<WeakRef<U>>
    where
        U: ScriptClass,
    {
        self.is_a::<U>()
          .then(||
            unsafe { mem::transmute::<&red::WeakHandle<red::ISerializable>, &WeakRef<U>>(&self.0.ref_) }
              .clone()
          )
    }
```